Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -33,9 +33,11 @@ internal class ConnectionPool : LoggerBase, IConnectionPool
         private readonly int _maxPoolSize;
         private readonly int _idlePoolSize;
         private readonly TimeSpan _connAcquisitionTimeout;
-        private readonly ConnectionValidator _connectionValidator;
+        internal TimeSpan ConnAcquisitionBlockingTimeout { get; set; } = TimeSpan.FromSeconds(5);
```

Comment:
Do we need to introduce a new internal timeout setting that could not be configured by the end user? I think that we should either favor ConnectionAcquisitionTimeOut value instead of this or fine tune default values (ConnectionAcquisitionTimeOut is set as 1m whereas this is 5m). What do you think?

Reply:
Here is 5s, which is the each small blocking timeout. So the idea was that when we blocking on waiting for a connection available from pool (here is 1m), there are two ways that we could claim a connection from pool: 1) waiting for pool to give us connection that is newly returned to the pool, 2) create a new one. This 5s timeout applies for how long we will block on 1) each time. In totally we will still blocking for 1mins, just we break every 5s that we will check if we could either do 2) or 1)

Reply:
I totally read the defaults wrong, sorry. But in the end, if the user sets the ConnectionAcquisitionTime to something around 0-10 seconds, I guess this internal configured value can cause a problem.

Reply:
I did a silly work around for your point that `ConnAcquisitionBlockingTimeout` will be reset to `AcquisitionTimeout` if `ConnAcquisitionBlockingTimeout > AcquisitionTimeout`. Pls also confirm with others what would be the best choice here.